### PR TITLE
app-emacs/calfw: Added patch for calfw time-ranges representation

### PR DESCRIPTION
**Pull request by lima-limon-inc**
---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

Sidenotes:
[calfw](https://github.com/kiwanami/emacs-calfw) Sadly, has not been updated for a long time ([Latest commit was 7 years ago](https://github.com/kiwanami/emacs-calfw/commit/03abce97620a4a7f7ec5f911e669da9031ab9088).)

I realized that calfw struggled to show org-mode events with time-ranges (Describe [here](https://orgmode.org/manual/Timestamps.html#index-time-range) in the org manual).
For example:
```
* TODO Timerange example
<2024-09-07 Wed>-<2024-09-11 Sun>
```

This event would be shown like this: 

No-patch

However, @MaximeWack made a pull request (here: https://github.com/kiwanami/emacs-calfw/pull/134) with the patch that I added to this PR.
With @MaximeWack 's patch, now that org mode event is shown correctly:

with-patch

I am not entirely familiar with what's Gentoo's standard procedure for when it comes to upstream fixes.

However, there doesn't appear to be a newer, more up-to-date calfw fork. 
And, calfw is a very neat package, specially for those of us who like to use Emacs a calendar; this time-range feature is very important.

Moreover, the patch, all things considered, is pretty small. 

Would love to hear some other perspectives.

### CODE REVIEW

**Comment by ulm** [comment]
Can you squash these patches into one?

**Comment by lima-limon-inc** [comment]
> Can you squash these patches into one?

Hello @ulm . Thanks for the feedback ^_^
I updated the patch in my latest commit. 
Is the format correct? 

Thanks in advance

**Comment by ulm** [comment]
Sorry, by "squash" I meant to combine all patches into one, similar to what the `squash` subcommand of `git rebase -i` would do.

It's only a minor point, so if it's too much of a hassle, then it is o.k. as-is (however, please keep _all_ commit messages from the original patch).

File: app-emacs/calfw/files/calfw-time-ranges.patch [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
++				 start-date
++				 (seconds-to-time (* 3600 24 (- total-days 1))))))
++		       (unless (= cur-day total-days)
++             (list (calendar-gregorian-from-absolute (time-to-days start-date))
++		                  (calendar-gregorian-from-absolute (time-to-days end-date)) text)))
+ 	     )))))
+
+ (defun cfw:org-schedule-period-to-calendar (begin end)
+--
+2.44.2
```
Comment (lima-limon-inc) [inline app-emacs/calfw/files/calfw-time-ranges.patch:40]: `git format-patch` added these lines at the end.
I do not know what those lines mean however.

Nevertheless, I tested the patch and the patch is applied correctly.

**Comment by lima-limon-inc** [comment]
Humble ping @ulm

File: app-emacs/calfw/calfw-1.6-r1.ebuild [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 1999-2019 Gentoo Authors
+# Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
```
Comment (xgqt) [inline app-emacs/calfw/calfw-1.6-r1.ebuild:4]: since we already rev-bump, let's also bump EAPI to 8

**Review by xgqt — CHANGES_REQUESTED** [review]

File: app-emacs/calfw/calfw-1.6-r1.ebuild (hunk 2 above)
Comment (lima-limon-inc) [reply app-emacs/calfw/calfw-1.6-r1.ebuild:4]: Sure thing! Will do so in next commit

**Comment by lima-limon-inc** [comment]
There we go! :D
What do you think @xgqt ?

**Review by xgqt — APPROVED** [review]

File: app-emacs/calfw/files/calfw-time-ranges.patch (hunk 1 above)
Comment (xgqt) [reply app-emacs/calfw/files/calfw-time-ranges.patch:40]: With this in src_prepare gives weird warning

```
>>> Preparing source in /var/tmp/portage/app-emacs/calfw-1.6-r1/work/emacs-calfw-1.6 ...
 * Applying calfw-time-ranges.patch ...
patch unexpectedly ends in middle of line                                                                   [ ok ]
>>> Source prepared.
```

If you remove those last lines it applies w/o warnings.

**Comment by ulm** [comment]
Pushed (with some adjustments to the patch file).
Thank you!

**Comment by lima-limon-inc** [comment]
> Pushed (with some adjustments to the patch file). Thank you!

^_^. Thank you both for all of you work! 
Have a good week.

Here's the commit if anyone from the future sees this PR: https://github.com/gentoo/gentoo/commit/717acbf6e8480ecca5df107092efaa766e745731

PS: What's the weather like in the future?

**Comment by xgqt** [comment]
@lima-limon-inc 

> PS: What's the weather like in the future?

Cool 23 deg. Celsius. For more discussion about weather and Lisp you can join us on #gentoo-lisp on Freenode IRC.

**Pull request closed without merging**